Statistical part for digit preference

LukaszPietrych opened the pull request.

LukaszPietrych committed:
Statistical part for digit preference

jonthegeek commented on R/statDigitPref.R:
```diff
@@ -0,0 +1,196 @@
+# ref.:
```
Should I ignore this file? It looks like `statDigitPref_refactoring.R` is at least closer to package-format.

jonthegeek commented on R/statDigitPref_refactoring.R:
```diff
@@ -0,0 +1,194 @@
+# ref.:
+
+# M. E. Stokes, C. S. Davis and G. G. Koch
+# Categorical Data Analysis Using the SAS System.
+# 3rd Edition.  Cary, NC: SAS Institute, Inc.
+
+# Kari Kuulasmaa, Hans-Werner Hense and Hanna Tolonen
+# Quality Assessment of Data on Blood Pressure in the WHO MONICA Project
+# MONICA Memo, May 1998
```
Leave this here, but we'll move it into the docs once we're done sorting out everything else.

jonthegeek commented on R/statDigitPref_refactoring.R:
```diff
@@ -0,0 +1,194 @@
+# Quality Assessment of Data on Blood Pressure in the WHO MONICA Project
+# MONICA Memo, May 1998
+
+# Installation of the function: gsm.digitpref
+pak::pak("OpenRBQM/gsm.digitpref")
+
+# Load necessary libraries
+library(dplyr)
+library(gsm.digitpref)
+library(pharmaversesdtm)
+library(stats)
+library(tidyr)
+
```
```suggestion
```
This is code within the package, so it can't install itself, and it can't `library()` packages. Whenever you reference a function from a package other than gsm.digitpref, provide the package namespace (`dplyr::`, etc). It looks like you mostly did that, but just to be sure. We also need to add those packages to the dependencies of this package. You can do that (relatively) easily by running this code in your R console:
```
usethis::use_package("dplyr")
usethis::use_package("tidyr")
```
To be clear, you shouldn't include those calls in the code you check in. You should run that code, which will generate changes to check in. Be sure to only do this for packages we actually use, though.

On the other hand, when something DOES come from gsm.digitpref, it's best *not* to namespace it (so `splitByPlace(`, not `gsm.digitpref::splitByPlace(`.

jonthegeek reviewed (changes requested):
We need a couple changes to get started. Clean these things up, and I should be able to sort out how to take it the rest of the way. If you're not sure what to do, let me know!

LukaszPietrych committed:
Update statDigitPref_refactoring.R

I have updated the code, revised the structure, and added relevant warnings. I believe your suggestions have been incorporated. Please let me know your thoughts on this version.

LukaszPietrych committed:
Delete R/statDigitPref.R
